File: clang/test/CodeGenCXX/debug-info-pubtypes.cpp
// REQUIRES: x86-registered-target
// RUN: %clang_cc1 -triple x86_64-apple-darwin10  -g -fstandalone-debug -S -mllvm -generate-dwarf-pub-sections=Enable %s -o - | FileCheck %s

// FIXME: This testcase shouldn't rely on assembly emission.
//CHECK: Lpubtypes_begin[[SECNUM:[0-9]:]]
//CHECK:         .asciz   "G"
//CHECK-NEXT:    .long   0
//CHECK-NEXT: Lpubtypes_end[[SECNUM]]

class G {
public:
  void foo();
};

void G::foo() {
}

File: clang/test/CodeGenObjC/debug-info-pubtypes.m
// REQUIRES: x86-registered-target
// RUN: %clang_cc1 -triple x86_64-apple-darwin10 -g -emit-llvm %s -o - | FileCheck %s

// CHECK: !MDCompositeType(tag: DW_TAG_structure_type, name: "H"
// CHECK-SAME:             line: [[@LINE+1]],
@interface H
-(void) foo;
@end

@implementation H
-(void) foo {
}
@end

